core/lib/test/unit/async-data-loader.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { AsyncDataLoader, AsyncDataLoaderAction } from '../../async-data-loader';
+
+jest.mock('@vue-storefront/core/lib/logger', () => ({
+  Logger: {
+    info: jest.fn(() => () => {})
+  }
+}))
+
+describe('AsyncDataLoader', () => {
+  it('contains new action with custom category parameter', () => {
+    const actionMock = {
+      execute: jest.fn(),
+      category: 'action1'
+    }
+
+    AsyncDataLoader.push(actionMock);
+
+    const expectedMock = {
+      execute: actionMock.execute,
+      category: 'action1',
+      scheduledAt: (actionMock as AsyncDataLoaderAction).scheduledAt
+    }
+
+    expect(AsyncDataLoader.queue[0].scheduledAt).toBe(expectedMock.scheduledAt)
+    expect(AsyncDataLoader.queue[0].category).toBe(expectedMock.category)
+    expect(AsyncDataLoader.queue[0].execute).toBe(expectedMock.execute)
+  })
+
+  it('contains new action with default category parameter', () => {
+    const actionMock = {
+      execute: jest.fn()
+    }
+
+    AsyncDataLoader.push(actionMock);
+
+    const expectedMock = {
+      execute: actionMock.execute,
+      category: 'asyncData',
+      scheduledAt: (actionMock as AsyncDataLoaderAction).scheduledAt
+    }
+
+    expect(AsyncDataLoader.queue[1].scheduledAt).toBe(expectedMock.scheduledAt)
```

Comment:
`queue[1]` - you cannot base on state from other test. If you would replace order of this test with one above - it will break. You have to take care of resetting state in each test. In this case, you have to set `queue` to `[]` at the very beginning of each test.